### PR TITLE
Verify build is clean

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

* Add .coverprofile to ignored files, so that travis builds do not mark binaries as "dirty"
* Add a basic check which ensures that the binaries are "clean" before pushing to the registry

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
Introduced an issue. will update the PR soon.

File: hack/verify-build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -e
+
+if [ -n "$(cluster/virtctl.sh version | grep dirty)" ]; then
```
Comment (davidvossel) [inline hack/verify-build.sh]: without a cluster being deployed, this is going to have a non-zero exit status, right?  

it woudl still convey the client's version correctly even without access to a cluster though.

Comment (rmohr) [reply hack/verify-build.sh]: Yes. I also have to add a positive check (that "clean" appears).

**Comment by rmohr** [comment]
@davidvossel @nellyc should be correct now.

**Pull request merged**